packages/ui/src/molecules/choose-modal/nft-modal.tsx

```diff
@@ -0,0 +1,168 @@
+        nft.getTokenName().toLowerCase().includes(searchInput.toLowerCase()),
+    )
+  }, [nftsOptions, searchInput])
+
+  useIntersectionObserver(itemRefs.current, async (index) => {
+    const originalIndex = options.findIndex(
+      (nft) => nft === filteredNfts[index],
+    )
+
+    if (originalIndex !== -1 && !options[originalIndex].isInited()) {
+      const updatedNft = await options[originalIndex].init()
+
+      setNftsOptions((prev) => {
+        const newOptions = prev.map((nft, i) =>
+          i === originalIndex ? updatedNft : nft,
+        )
+        return [...new Set(newOptions)]
+      })
+    }
+  })
+
+  const handleSelect = useCallback((token: NFT) => {
+    setSelectedValue(token.getTokenId())
+
+    setTimeout(
+      () => {
+        setIsModalVisible(false)
+      },
+      isSmooth ? 100 : 0,
+    )
+  }, [])
+
+  useEffect(() => {
+    onSelect && onSelect(selectedValue)
+  }, [selectedValue])
+
+  return (
+    <div className="flex flex-col shrink-0" id={"choose_modal"}>
+      {label && <Label className="mb-1">{label}</Label>}
+      <div className="flex shrink-0" onClick={() => setIsModalVisible(true)}>
+        {trigger}
+      </div>
+      <div
+        className={clsx(
+          "p-5 absolute w-full h-full z-50 left-0 top-0 bg-frameBgColor",
+          "flex flex-col rounded-xl",
+          !isModalVisible && "hidden",
+        )}
+      >
+        <div className="flex justify-between">
+          <div className="flex items-center">
+            <div
+              className="cursor-pointer"
+              onClick={() => setIsModalVisible(false)}
+            >
+              <IconCmpArrow className="mr-2" />
+            </div>
+            <p className="text-xl font-bold leading-10">{title}</p>
+          </div>
+          {warningText && (
+            <Tooltip tip={warningText}>
+              <IconCmpWarning className="cursor-help hover:opacity-70 text-orange" />
+            </Tooltip>
+          )}
+        </div>
+        <Input
+          type="text"
+          placeholder="Search"
+          inputClassName="!border-black"
+          icon={<IoIosSearch size="20" className="text-gray-400" />}
+          onKeyUp={(e) => setSearchInput((e.target as HTMLInputElement).value)}
+          className="mt-4 mb-5"
+        />
+        {isNftOptionsLoading && <ChooseTokenSkeleton rows={6} />}
+        <div
+          className={clsx(
+            "flex-1 overflow-auto snap-end pr-[10px]",
+            "scrollbar scrollbar-w-4 scrollbar-thumb-gray-300",
+            "scrollbar-thumb-rounded-full scrollbar-track-rounded-full",
+          )}
+        >
+          {filteredNfts.map((nft, index) => (
+            <div
+              id={`option_${nft.getTokenName().replace(/\s/g, "")}`}
+              key={`group_${nft.getCollectionId()}_${nft.getTokenName()}_${index}`}
+              ref={(el) => (itemRefs.current[index] = el)}
+            >
+              <ChooseNftItem
```

Comment:
if these two modals only differ in option component being rendered it definitely makes sense to not duplicate such a big piece of code, you could have one component for this purpose and pass ChooseNftItem as an optionComponent prop to modal